spec.bs

```diff
@@ -1093,21 +1094,32 @@ Issue: Finish specifying the following algorithms.
       1. Let |componentDescriptor| be a new [=ad descriptor=].
       1. If |component| is {{DOMString}}:
         1. Let |componentUrl| be the result of running the [=URL parser=] on |component|.
-        1. If |componentUrl| is an error, return false.
-        1. If [=validating an ad url=] given |componentUrl| returns false, return false.
+        1. If |componentUrl| is an error, return failure.
+        1. If [=validating an ad url=] given |componentUrl| returns false, return failure.
         1. Set |componentDescriptor|'s [=ad descriptor/url=] to |componentUrl|.
       1. Otherwise:
         1. Set |componentDescriptor| to the result of [=converting an ad render=] given |component|.
-        1. If |componentDescriptor| is failure, return false.
+        1. If |componentDescriptor| is failure, return failure.
       1. [=list/Append=] |componentDescriptor| to |adComponentDescriptors|.
-    1. Set |bidToSet|'s [=generated bid/ad component descriptors=] to |adComponentDescriptors|.
+    1. Set |bid|'s [=generated bid/ad component descriptors=] to |adComponentDescriptors|.
   1. If |generateBidOutput|["{{GenerateBidOutput/adCost}}"] [=map/exists=]:
-    1. Set |bidToSet|'s [=generated bid/ad cost=] to
+    1. Set |bid|'s [=generated bid/ad cost=] to
       |generateBidOutput|["{{GenerateBidOutput/adCost}}"].
   1. If |generateBidOutput|["{{GenerateBidOutput/modelingSignals}}"] [=map/exists=]:
     1. Let |modelingSignals| be |generateBidOutput|["{{GenerateBidOutput/modelingSignals}}"].
     1. If |modelingSignals| is greater than or equal to 0 and less than 4096:
-      1. Set |bidToSet|'s [=generated bid/modeling signals=] to |modelingSignals|
+      1. Set |bid|'s [=generated bid/modeling signals=] to |modelingSignals|
+  1. Return |bid|.
+
+</div>
+
+<div algorithm>
+  The <dfn method for="InterestGroupBiddingScriptRunnerGlobalScope">setBid(|generateBidOutput|)</dfn>
+  method steps are:
+
+  1. Clear the bid. TODO: expand.
+  1. Let |bidToSet| be the result of [=converting GenerateBidOutput to generated bid=] with |generateBidOutput|.
+  1. If |bidToSet| is failure, return false.
   1. Set bid to |bidToSet|.
   1. Return true.
 </div>

```

Comment:
Do developers still need to use the `setBid()` function anywhere, if they can return their bid from the `generateBid()` function?